numpy/_core/src/multiarray/item_selection.c

```diff
@@ -2896,7 +2896,8 @@ PyArray_Nonzero(PyArrayObject *self)
             if (((double)nonzero_count / count) <= 0.1) {
                 npy_intp subsize;
                 npy_intp j = 0;
-                while (1) {
+                npy_intp * multi_index_end = multi_index + nonzero_count;
```

Comment:
It looks like this is defined in the same way in the `else` branch - can you define it outside the if/else once?